src/devices/Media/VideoDevice/Devices/UnixVideoDevice.cs

```diff
@@ -0,0 +1,546 @@
+            ctrl.id = VideoDeviceValueType.Brightness;
+            ctrl.value = Settings.Brightness;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set contrast
+            ctrl.id = VideoDeviceValueType.Contrast;
+            ctrl.value = Settings.Contrast;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set saturation
+            ctrl.id = VideoDeviceValueType.Saturation;
+            ctrl.value = Settings.Saturation;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set sharpness
+            ctrl.id = VideoDeviceValueType.Sharpness;
+            ctrl.value = Settings.Sharpness;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set gain
+            ctrl.id = VideoDeviceValueType.Gain;
+            ctrl.value = Settings.Gain;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set gamma
+            ctrl.id = VideoDeviceValueType.Gamma;
+            ctrl.value = Settings.Gamma;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set power line frequency
+            ctrl.id = VideoDeviceValueType.PowerLineFrequency;
+            ctrl.value = (int)Settings.PowerLineFrequency;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set white balance effect
+            ctrl.id = VideoDeviceValueType.WhiteBalanceEffect;
+            ctrl.value = (int)Settings.WhiteBalanceEffect;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set white balance temperature
+            ctrl.id = VideoDeviceValueType.WhiteBalanceTemperature;
+            ctrl.value = Settings.WhiteBalanceTemperature;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set color effect
+            ctrl.id = VideoDeviceValueType.ColorEffect;
+            ctrl.value = (int)Settings.ColorEffect;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set scene mode
+            ctrl.id = VideoDeviceValueType.SceneMode;
+            ctrl.value = (int)Settings.SceneMode;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set rotate
+            ctrl.id = VideoDeviceValueType.Rotate;
+            ctrl.value = Settings.Rotate;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set horizontal flip
+            ctrl.id = VideoDeviceValueType.HorizontalFlip;
+            ctrl.value = Settings.HorizontalFlip ? 1 : 0;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+
+            // Set vertical flip
+            ctrl.id = VideoDeviceValueType.VerticalFlip;
+            ctrl.value = Settings.VerticalFlip ? 1 : 0;
+            V4l2Struct(VideoSettings.VIDIOC_S_CTRL, ref ctrl);
+        }
+
+        private void FillVideoConnectionSettings()
+        {
+            if (Settings.CaptureSize.Equals(default))
```

Comment:
Should this return settings instead?

Reply:
The logic now is to modify the property. If the user does not set this property, the method reads the default setting from the v4l2 driver.